nengo_spa/modules/assoc_mem.py

```diff
@@ -31,10 +34,12 @@ class AssociativeMemory(Network):
         The vocabulary to be produced for each match. If
         None, the associative memory will act like an autoassociative memory
         (cleanup memory).
-    input_keys : list, optional (Default: None)
-        A list of strings that correspond to the input vectors.
-    output_keys : list, optional (Default: None)
-        A list of strings that correspond to the output vectors.
+    mapping: dict or str, optional (Default: None)
+        A dictionary that defines the mapping from Semantic Pointers in the
+        input vocabulary to Semantic Pointers in the output vocabulary. If set
+        to the string ``'by-key'``, the mapping will be done based on the keys
```

Comment:
I'm thinking whether it's worth changing 'by-key' to just 'key'. I'd say it's a bit easier to write. This is just a suggestion so you can also leave 'by-key' if you like that better